core/src/test/scala/org/apache/spark/rdd/RDDSuite.scala

```diff
@@ -726,8 +725,8 @@ class RDDSuite extends FunSuite with SharedSparkContext {
 
     val repartitioned = data.repartitionAndSortWithinPartitions(partitioner)
     val partitions = repartitioned.glom().collect()
-    assert(partitions(0) === Seq((0, 5), (0, 8), (2, 6)))
-    assert(partitions(1) === Seq((1, 3), (3, 8), (3, 8)))
+    assert(partitions(0).toSet === Set((0, 5), (0, 8), (2, 6)))
+    assert(partitions(1).toSet === Set((1, 3), (3, 8), (3, 8)))
```

Comment:
Hi @adrian-wang , the ordering is correct, since we only guarantee by-key ordering, the order of tuples with same key is not guaranteed by this PR, this required secondary-sort.